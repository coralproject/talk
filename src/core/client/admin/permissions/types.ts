import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
