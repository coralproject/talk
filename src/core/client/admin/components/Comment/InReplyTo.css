$moderateCardIconColor: var(--palette-text-100);
$moderateCardInReplyToColor: var(--palette-text-100);
$moderateCardInReplyToUsernameColor: var(--palette-text-500);
$moderateCardUsernameBackgroundHover: var(--palette-grey-200);

.icon {
  color: $moderateCardIconColor;
}
.inReplyTo {
  color: $moderateCardInReplyToColor;
  font-family: var(--font-family-primary);
  font-weight: var(--font-weight-primary-regular);
  font-size: var(--font-size-2);
  line-height:  1.14;
}
.username {
  color: $moderateCardInReplyToUsernameColor;
  font-family: var(--font-family-secondary);
  font-weight: var(--font-weight-secondary-bold);
  font-size: var(--font-size-3);
  line-height: 1;

  padding: var(--spacing-1);

  &:hover {
    background-color: $moderateCardUsernameBackgroundHover;
    border-radius: var(--round-corners);
    border-style: none;
  }

  &:active {
    background-color: $moderateCardUsernameBackgroundHover;
    border-radius: var(--round-corners);
    border-style: none;
  }
}

.usernameButton {
  vertical-align: baseline;

  margin-left: calc(-1 * var(--spacing-1));
}
