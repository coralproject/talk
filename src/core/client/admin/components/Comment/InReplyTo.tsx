import { Localized } from "@fluent/react/compat";
import React, { FunctionComponent } from "react";

import { BaseButton, Flex, Icon } from "coral-ui/components/v2";

import styles from "./InReplyTo.css";

interface Props {
  children: string;
  onUsernameClick: () => void;
}

const InReplyTo: FunctionComponent<Props> = ({ children, onUsernameClick }) => {
  const Username = () => (
    <BaseButton onClick={onUsernameClick} className={styles.usernameButton}>
      <span className={styles.username}>{children}</span>
    </BaseButton>
  );

  return (
    <Flex alignItems="center">
      <Icon className={styles.icon}>reply</Icon>{" "}
      <Localized id="moderate-comment-inReplyTo" Username={<Username />}>
        <span className={styles.inReplyTo}>
          Reply to
          <Username />
        </span>
      </Localized>
    </Flex>
  );
};

export default InReplyTo;
