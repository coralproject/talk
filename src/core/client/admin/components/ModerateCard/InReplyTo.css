$moderateCardIconColor: var(--v2-colors-mono-100);
$moderateCardInReplyToColor: var(--v2-colors-mono-100);
$moderateCardInReplyToUsernameColor: var(--v2-colors-mono-500);
$moderateCardUsernameBackgroundHover: var(--v2-colors-grey-200);

.icon {
  color: $moderateCardIconColor;
}
.inReplyTo {
  color: $moderateCardInReplyToColor;
  font-family: var(--v2-font-family-primary);
  font-weight: var(--v2-font-weight-primary-regular);
  font-size: var(--v2-font-size-2);
  line-height: var(--v2-line-height-reset);
}
.username {
  color: $moderateCardInReplyToUsernameColor;
  font-family: var(--v2-font-family-secondary);
  font-weight: var(--v2-font-weight-secondary-bold);
  font-size: var(--v2-font-size-3);
  line-height: var(--v2-line-height-reset);

  padding: var(--spacing-1);

  &:hover {
    background-color: $moderateCardUsernameBackgroundHover;
    border-radius: var(--round-corners);
    border-style: none;
  }

  &:active {
    background-color: $moderateCardUsernameBackgroundHover;
    border-radius: var(--round-corners);
    border-style: none;
  }
}

.usernameButton {
  vertical-align: baseline;

  margin-left: calc(-1 * var(--v2-spacing-1));
}
