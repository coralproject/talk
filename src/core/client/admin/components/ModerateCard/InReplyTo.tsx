import { Localized } from "fluent-react/compat";
import React, { FunctionComponent } from "react";

import { Flex, Icon, Typography } from "coral-ui/components";

import styles from "./InReplyTo.css";

interface Props {
  children: string;
}

const InReplyTo: FunctionComponent<Props> = ({ children }) => {
  const Username = () => (
    <Typography variant="heading5" container="span" className={styles.username}>
      {children}
    </Typography>
  );

  return (
    <Flex alignItems="center">
      <Icon className={styles.icon}>reply</Icon>{" "}
      <Localized id="moderate-comment-inReplyTo" Username={<Username />}>
        <Typography
          variant="timestamp"
          container="span"
          className={styles.inReplyTo}
        >
          {"Reply to <Username></Username>"}
        </Typography>
      </Localized>
    </Flex>
  );
};

export default InReplyTo;
