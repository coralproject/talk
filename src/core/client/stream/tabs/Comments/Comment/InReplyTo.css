.icon {
  color: var(--palette-grey-light);
}
.inReplyTo {
  color: var(--palette-grey-light);
}
.username {
  color: var(--palette-grey-dark);
}
