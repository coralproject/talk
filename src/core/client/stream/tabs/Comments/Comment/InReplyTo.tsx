import { Localized } from "@fluent/react/compat";
import cn from "classnames";
import React, { FunctionComponent } from "react";

import CLASSES from "coral-stream/classes";
import { Flex, Icon } from "coral-ui/components/v2";

import styles from "./InReplyTo.css";

interface Props {
  username: string;
}

const InReplyTo: FunctionComponent<Props> = ({ username }) => {
  const Username = () => (
    <span className={cn(styles.username, CLASSES.comment.inReplyTo.username)}>
      {username}
    </span>
  );

  return (
    <Flex alignItems="center" className={CLASSES.comment.inReplyTo.$root}>
      <Icon className={styles.icon}>reply</Icon>{" "}
      <Localized id="comments-inReplyTo" Username={<Username />}>
        <span className={cn(styles.inReplyTo, CLASSES.comment.inReplyTo.text)}>
          {"In reply to <Username></Username>"}
        </span>
      </Localized>
    </Flex>
  );
};

export default InReplyTo;
