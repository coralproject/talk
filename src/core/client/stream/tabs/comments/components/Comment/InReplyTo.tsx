import { Localized } from "fluent-react/compat";
import React, { StatelessComponent } from "react";

import { Flex, Icon, Typography } from "talk-ui/components";

import styles from "./InReplyTo.css";

interface Props {
  username: string;
}

const InReplyTo: StatelessComponent<Props> = ({ username }) => {
  const Username = () => (
    <Typography variant="heading5" container="span" className={styles.username}>
      {username}
    </Typography>
  );

  return (
    <Flex alignItems="center">
      <Icon className={styles.icon}>reply</Icon>{" "}
      <Localized id="comments-inReplyTo" username={<Username />}>
        <Typography
          variant="timestamp"
          container="span"
          className={styles.inReplyTo}
        >
          {"In reply to <username><username>"}
        </Typography>
      </Localized>
    </Flex>
  );
};

export default InReplyTo;
