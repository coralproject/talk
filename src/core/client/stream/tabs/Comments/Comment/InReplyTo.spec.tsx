import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "coral-framework/types";

import InReplyTo from "./InReplyTo";

it("renders correctly", () => {
  const props: PropTypesOf<typeof InReplyTo> = {
    parent: {
      id: "test-id",
      author: {
        username: "Username",
      },
    },
    enableJumpToParent: false,
  };
  const renderer = createRenderer();
  renderer.render(<InReplyTo {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
