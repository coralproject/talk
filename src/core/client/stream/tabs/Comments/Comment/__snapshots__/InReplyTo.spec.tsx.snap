// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(forwardRef)
  alignItems="center"
  className="coral coral-comment-inReplyTo"
>
  <ForwardRef(forwardRef)
    className="InReplyTo-icon"
  >
    reply
  </ForwardRef(forwardRef)>
   
  <Localized
    Username={<Username />}
    id="comments-inReplyTo"
  >
    <span
      className="InReplyTo-inReplyTo coral coral-comment-inReplyToText"
    >
      In reply to &lt;Username&gt;&lt;/Username&gt;
    </span>
  </Localized>
</ForwardRef(forwardRef)>
`;
